integration/src/main/java/io/opencensus/interop/TestUtils.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.interop;
+
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+import io.opencensus.contrib.http.util.HttpPropagationUtil;
+import io.opencensus.tags.InternalUtils;
+import io.opencensus.tags.Tag;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.Tags;
+import io.opencensus.tags.propagation.TagContextBinarySerializer;
+import io.opencensus.tags.propagation.TagContextDeserializationException;
+import io.opencensus.tags.propagation.TagContextSerializationException;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.propagation.TextFormat;
+import java.math.BigInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Interop test utilities. */
+public class TestUtils {
+
+  private static final Logger logger = Logger.getLogger(TestUtils.class.getName());
+
+  private TestUtils() {}
+
+  /**
+   * Try to get port configuration from the environment and if not specified, use the default one.
+   *
+   * @param varName the environment variable name.
+   * @return the port.
+   */
+  public static int getPortOrDefault(String varName, int defaultPort) {
+    int portNumber = defaultPort;
+    String portStr = System.getenv(varName);
+    if (portStr != null) {
+      try {
+        portNumber = Integer.parseInt(portStr);
+      } catch (NumberFormatException e) {
+        logger.warning(
+            String.format("Port %s is invalid, use default port %d.", portStr, defaultPort));
+      }
+    }
+    return portNumber;
+  }
+
+  /**
+   * Build the {@link io.opencensus.interop.EchoResponse} according to the given {@link SpanContext}
+   * and {@link TagContext}.
+   *
+   * @param spanContext the {@code SpanContext}. Can be {@code null}.
+   * @param tagContext the {@code TagContext}. Can be {@code null}.
+   * @return a {@code EchoResponse}.
+   */
+  public static EchoResponse buildResponse(SpanContext spanContext, TagContext tagContext) {
+    TagContextBinarySerializer serializer = Tags.getTagPropagationComponent().getBinarySerializer();
+    EchoResponse.Builder builder = EchoResponse.newBuilder();
+
+    try {
+      if (spanContext != null) {
```

Comment:
For now, there is no such case that `SpanContext` will be null.
I leave the logic here just in case we only want to check `TagContext` in the future. The idea is pre-mature. I can remove it and if we ever need it, it can be added back.

Reply:
Sounds good, I'm fine with leaving it as is.